bokehjs/src/lib/models/widgets/tables/row_aggregators.ts

```diff
@@ -0,0 +1,96 @@
+const { Aggregators: { Avg, Min, Max, Sum } } = require('slickgrid/slick.dataview')
+
+import * as p from 'core/properties'
+import { Model } from 'model'
+
+export type GroupTotals = {
+  avg: { [field_: string]: number }
+  min: { [field_: string]: number }
+  max: { [field_: string]: number }
+  sum: { [field_: string]: number }
+}
+
+export namespace RowAggregator {
+  export type Attrs = p.AttrsOf<Props>
+
+  export type Props = Model.Props & {
+    field_: p.Property<string>
+  }
+}
+
+export interface RowAggregator extends RowAggregator.Attrs {
+  readonly key: string
+}
+
+export abstract class RowAggregator extends Model {
+  properties: RowAggregator.Props
+
+  constructor(attrs?: Partial<RowAggregator.Attrs>) {
+      super(attrs)
+  }
+
+  static initClass(): void {
+    this.prototype.type = 'RowAggregator'
+
+    this.define<RowAggregator.Props>({
+      field_: [ p.String, '' ],
+    })
+  }
+
+  abstract init(): void
+  abstract accumulate(item: { [key: string]: any }): void
+  abstract storeResult(totals: GroupTotals): void
+}
+RowAggregator.initClass()
+
+export class AvgAggregator extends RowAggregator {
+  readonly key = 'avg'
+
+  static initClass(): void {
+    this.prototype.type = 'AvgAggregator'
+  }
+
+  init = new Avg().init
+  accumulate = new Avg().accumulate
+  storeResult = new Avg().storeResult
```

Comment:
Unless it's mandated by an external API, all variables and member names should use snake case.

Reply:
Three instances of `Avg` per `AvgAggregator` is an excess, and shouldn't work if there is any state involved. A more robust approach would be to store private `Avg` instance and implement relevant abstract methods explicitly to work on that instance.